### ISSUE TITLE
Method to reserve a connection without starting a transaction

### ISSUE THREAD

**Issue by valeneiko** [porsager/postgres#616]
# Summary

I would like to be able to reserve a connection from a pool without starting a transaction:
```js
const sql = postgres(...);
await sql.connect(async (conn) => {
  await conn`SELECT 1`;
  await conn`SELECT 2`;
});
```

# Use case
We have multiple database replicas and after each write operation we verify the write is visible on the replicas by querying the state of the replica. The write should not happen if we are unable to connect to replica.

Something like this:
```js
const main = postgres(...);
const repl = postgres(...);

// Get the connection to the replica, to ensure we can verify it later
await repl.connect(async (conn) => {
  // Do the write and return transaction ID
  const [{tx_id}] = await main.transaction(async (tx) => {
    await tx`UPDATE test(x) SET ...`;
    return tx`SELECT get_tx_id() as tx_id`;
  });
  
  // Verify write is visible on the replica
  while(true) {
    const [{is_visible}] = await conn`SELECT is_visible(${tx_id})`;
    if(is_visible) {
      break;
    }
    await setTimeout(100);
  }
}
```

**Comment by porsager** [comment]
I think this might be a duplicate of https://github.com/porsager/postgres/issues/603 ?

But for your usecase, I'm wondering - could the [multi-host connection setup](https://github.com/porsager/postgres#multi-host-connections---high-availability-ha) isn't working for you?

**Comment by valeneiko** [comment]
Yes it's a duplicate, sorry about that!

As for multi-host I am not sure. We use pool instances so that we can send all read queries to replicas and only send write queries to main. I didn't think we can configure multi-host to do that, can we?

But even then, how would I ensure read connection is available before starting the write?

**Comment by porsager** [comment]
I think you'd do the following (only minor edits to your code):

```
const options = {
  host: ['host1', 'host2'],
  ...otherOptions
}

const main = postgres({ ...options, target_session_attrs: 'read-write' })
const repl = postgres({ ...options, target_session_attrs: 'read-only' })

await repl.begin(repl => {
  const [{ tx_id }] = await main.begin(main => {
    await main`UPDATE test(x) SET ...`
    return main`SELECT get_tx_id() as tx_id`
  })
  
  while(true) {
    const [{is_visible}] = await repl`SELECT is_visible(${tx_id})`
    if(is_visible)
      break
    await setTimeout(100)
  }
}
```

Let me know if that works out for you.

**Comment by valeneiko** [comment]
That works.
